src/Command/Ingester/LtiInstanceIngesterCommand.php

```diff
@@ -0,0 +1,151 @@
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2020 (original work) Open Assessment Technologies S.A.
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\Command\Ingester;
+
+use OAT\SimpleRoster\Csv\CsvReaderBuilder;
+use OAT\SimpleRoster\Entity\LtiInstance;
+use OAT\SimpleRoster\Repository\LtiInstanceRepository;
+use OAT\SimpleRoster\Storage\StorageRegistry;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Throwable;
+
+class LtiInstanceIngesterCommand extends AbstractCsvIngesterCommand
+{
+    public const NAME = 'roster:ingest:lti-instance';
+
+    /** @var LtiInstanceRepository */
+    private $ltiInstanceRepository;
+
+    public function __construct(
+        CsvReaderBuilder $csvReaderBuilder,
+        StorageRegistry $storageRegistry,
+        LtiInstanceRepository $ltiInstanceRepository
+    ) {
+        parent::__construct($csvReaderBuilder, $storageRegistry);
+
+        $this->ltiInstanceRepository = $ltiInstanceRepository;
+    }
+
+    protected function getIngesterCommandName(): string
+    {
+        return self::NAME;
+    }
+
+    protected function configure(): void
+    {
+        parent::configure();
+
+        $this->setDescription('LTI instance ingestion from various sources');
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        parent::initialize($input, $output);
+
+        $this->symfonyStyle->title('Simple Roster - LTI Instance Ingester');
+    }
+
+    /**
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->symfonyStyle->text('Executing ingestion...');
+        $this->symfonyStyle->newLine();
+
+        try {
+            $this->progressBar->start();
+
+            $numberOfProcessedRows = 0;
+            $persisted = false;
+            foreach ($this->csvReader->getRecords() as $rawLtiInstance) {
+                $this->validateRow($rawLtiInstance, 'label', 'ltiLink', 'ltiKey', 'ltiSecret');
+
+                $numberOfProcessedRows++;
+                $persisted = false;
+
+                $ltiInstance = new LtiInstance(
+                    0,
+                    $rawLtiInstance['label'],
+                    $rawLtiInstance['ltiLink'],
+                    $rawLtiInstance['ltiKey'],
+                    $rawLtiInstance['ltiSecret']
+                );
+
+                $this->ltiInstanceRepository->persist($ltiInstance);
+
+                if ($numberOfProcessedRows % $this->batchSize === 0) {
```

Comment:
Can we isolate this conditional? The reason is that it is showing up several times and to better explain what it means.

Reply:
It is not showing up several times, only once per command. I see your point about explaining it, but I don't know how to make it more clear. Any suggestions?

Reply:
Fixed